src/components/TransactionBlock/WalletTransactionBlock.tsx

```diff
@@ -0,0 +1,737 @@
+        if (!chainAsset.assets || !selectedChains.includes(chainAsset.chainId)) return;
+
+        chainAsset.assets.map((asset) => {
+          if (
+            !!searchValue &&
+            !asset.name.toLowerCase().includes(searchValue.toLowerCase()) &&
+            !asset.symbol.toLowerCase().includes(searchValue.toLowerCase())
+          )
+            return;
+
+          assets.push({ ...asset, chainId: chainAsset.chainId });
+        });
+      });
+    }
+
+    setDisplayAssets(assets);
+  }, [chainAssets, selectedChains, showAllChains, searchValue]);
+
+  const findTransactionBlock = (blockType: TRANSACTION_BLOCK_TYPE_KEY) => {
+    return availableTransactionBlocks.find((item) => item?.type === blockType) || null;
+  };
+
+  const handleAddTransaction = (availableTransactionBlock: ITransactionBlock | null) => {
+    if (!availableTransactionBlock) return;
+
+    const isBridgeTransactionBlock = availableTransactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE;
+    const isBridgeTransactionBlockAndDisabled = isBridgeTransactionBlock && hasTransactionBlockAdded;
+    const isKlimaBlockIncluded = availableTransactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE;
+
+    addTransactionBlock(
+      availableTransactionBlock,
+      isBridgeTransactionBlockAndDisabled || false,
+      isKlimaBlockIncluded || false
+    );
+  };
+
+  const handleActionButton = (blockType: TRANSACTION_BLOCK_TYPE_KEY) => {
+    const block = findTransactionBlock(blockType);
+    handleAddTransaction(block);
+  };
+
+  const handleDepositButton = () => {
+    const block = findTransactionBlock(TRANSACTION_BLOCK_TYPE.SEND_ASSET);
+    if (!block || block.type !== TRANSACTION_BLOCK_TYPE.SEND_ASSET) return;
+
+    if (!providerAddress || !accountAddress) return;
+
+    let values: ISendAssetTransactionBlockValues = {};
+    values.fromAddress = providerAddress;
+    values.receiverAddress = accountAddress;
+    values.isFromEtherspotWallet = false;
+
+    block.values = values;
+    handleAddTransaction(block);
+  };
+
+  const handleSendButton = () => {
+    const block = findTransactionBlock(TRANSACTION_BLOCK_TYPE.SEND_ASSET);
+    if (!block || block.type !== TRANSACTION_BLOCK_TYPE.SEND_ASSET) return;
+
+    if (!providerAddress || !accountAddress) return;
+
+    let values: ISendAssetTransactionBlockValues = {};
+    values.fromAddress = accountAddress;
+    values.isFromEtherspotWallet = true;
+
+    block.values = values;
+    handleAddTransaction(block);
+  };
+
+  const handleWithdrawButton = () => {
+    const block = findTransactionBlock(TRANSACTION_BLOCK_TYPE.SEND_ASSET);
+    if (!block || block.type !== TRANSACTION_BLOCK_TYPE.SEND_ASSET) return;
+
+    if (!providerAddress || !accountAddress) return;
+
+    let values: ISendAssetTransactionBlockValues = {};
+    values.fromAddress = accountAddress;
+    values.receiverAddress = providerAddress;
+    values.isFromEtherspotWallet = true;
+
+    block.values = values;
```

Comment:
hm, please make sure `block` does not mutate in all these handlers

Reply:
it didn't during testing, but i've added a new variable just in case